### ISSUE TITLE
Bug with inherited constructor and optional named parameter

### ISSUE THREAD

**Issue by maarekj** [vimeo/psalm#5656]
Hello,

If I declare  a **class A**, with 2 parameters in constructor: *lastname* and *firstname*.
And if I declare a **class B** whose inherit of A, with 2 parameters in constructor.

If one of the parameters is **optional and named**, a **bad call to the parent constructor** will not be detected.

https://psalm.dev/r/e492fb7855

Example:

```php
<?php

class A {
	public function __construct(string $lastname, string $firstname = "") {
    }
}

// bug
class B extends A {
    public function __construct(string $name, string $firstname = "") {
        self::__construct(firstname: $firstname);
    }
}

// ok
class C extends A {
    public function __construct(string $name, string $firstname = "") {
        self::__construct();
    }
}
```

```
Psalm output (using commit 0d524ec): 

ERROR: TooFewArguments - 18:15 - Too few arguments for C::__construct - expecting 1 but saw 0
```

On line 11, there is an error that **is not detected**.
On the other hand, on line 18, the error **is detected**.

Thanks you very much for your wonderful work.

**Comment by orklah** [comment]
This was fixed :)

**Comment by maarekj** [comment]
Thanks yoj
